components/script_traits/lib.rs

```diff
@@ -197,6 +198,9 @@ pub enum DiscardBrowsingContext {
 /// Messages sent from the constellation or layout to the script thread.
 #[derive(Deserialize, Serialize)]
 pub enum ConstellationControlMsg {
+    /// Sends the final request to script thread for fetching after all redirections
+    /// have been resolved
+    FinalRequest(PipelineId, RequestInit),
```

Comment:
This will be able to contain a `Result<FetchMetadata, ...>` instead of `RequestInit` if the constellation sends the final response header received instead. We can call it `NavigationResponse` instead